core/internal/runconsolelogs/filestreamwriter.go

```diff
@@ -11,34 +10,24 @@ import (
 // filestreamWriter sends modified console log lines to the filestream.
 type filestreamWriter struct {
 	FileStream filestream.FileStream
+
+	// StructuredLogs indicates whether to send structured logs.
+	Structured bool
 }
 
 func (w *filestreamWriter) SendChanged(
 	changes sparselist.SparseList[*RunLogsLine],
 ) {
-	// Generate timestamps in the Python ISO format (microseconds without Z)
-	const rfc3339Micro = "2006-01-02T15:04:05.000000Z07:00"
-
 	lines := sparselist.Map(changes, func(line *RunLogsLine) string {
-		timestamp := strings.TrimSuffix(
-			line.Timestamp.UTC().Format(rfc3339Micro), "Z")
-
-		if line.StreamLabel != "" {
-			return fmt.Sprintf(
-				"%s%s [%s] %s",
-				line.StreamPrefix,
-				timestamp,
-				line.StreamLabel,
-				string(line.Content),
-			)
-		} else {
-			return fmt.Sprintf(
-				"%s%s %s",
-				line.StreamPrefix,
-				timestamp,
-				string(line.Content),
-			)
+		if w.Structured {
+			jsonLine, err := json.Marshal(line)
+			if err != nil {
+				return err.Error()
```

Comment:
If there's a marshaling error, we upload the error instead of the line? That seems strange. We should log and possibly capture the error instead.

Reply:
Oops, didn't commit the change -- the best we can do is to fall back to the legacy format (in addition to logging the error).